daemon/logger/awslogs/cwlogsiface_mock_test.go

```diff
@@ -49,6 +60,13 @@ func (m *mockcwlogsclient) PutLogEvents(input *cloudwatchlogs.PutLogEventsInput)
 	return output.successResult, output.errorResult
 }
 
+func (m *mockcwlogsclient) CreateLogGroup(input *cloudwatchlogs.CreateLogGroupInput) (*cloudwatchlogs.CreateLogGroupOutput, error) {
+	// m.createLogGroupArgument <- input
```

Comment:
You can see an example here: https://github.com/docker/docker/blob/7a016578dbb5c9b8a43b43e1ce96785fa26baf56/daemon/logger/awslogs/cloudwatchlogs_test.go#L79-L85

This sets an expectation by way of giving the mock a result that it can return when it gets called.  You can then [read the argument(s)](https://github.com/docker/docker/blob/7a016578dbb5c9b8a43b43e1ce96785fa26baf56/daemon/logger/awslogs/cloudwatchlogs_test.go#L92-L95) and perform assertions.  It's somewhat crude since I hand-wrote it (I didn't see an existing mocking tool being used in Docker), but it seems to work.